src/react/src/components/ClaimFacilityIntroStep.jsx

```diff
@@ -1,79 +1,171 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import Typography from '@material-ui/core/Typography';
-import List from '@material-ui/core/List';
-import ListItem from '@material-ui/core/ListItem';
-import ListItemIcon from '@material-ui/core/ListItemIcon';
-import ListItemText from '@material-ui/core/ListItemText';
-import CheckIcon from '@material-ui/icons/Check';
+import FormControl from '@material-ui/core/FormControl';
+import FormControlLabel from '@material-ui/core/FormControlLabel';
+import RadioGroup from '@material-ui/core/RadioGroup';
+import Radio from '@material-ui/core/Radio';
 
-import { claimAFacilityFormStyles } from '../util/styles';
+import { connect } from 'react-redux';
+import { func } from 'prop-types';
+import BadgeClaimed from './BadgeClaimed';
+import { updateClaimFacilityIntro } from '../actions/claimFacility';
 
-import COLOURS from '../util/COLOURS';
+const radioGroupStyles = Object.freeze({
+    radioItem: {
+        marginRight: '30px',
+    },
+    radio: {
+        color: 'rgb(0, 0, 0)',
+        padding: '3px 5px 0 7px',
+    },
+});
 
-const introMessageStyles = Object.freeze({
-    width: '95%',
-    padding: '10px',
+const inputGroupStyles = Object.freeze({
+    width: '100%',
+    paddingTop: '30px',
 });
 
-const checkIconStyles = Object.freeze({
-    color: COLOURS.NAVY_BLUE,
+const boldStyle = Object.freeze({
+    fontWeight: 'bold',
+});
+const paragraphStyle = Object.freeze({
+    paddingBottom: '20px',
+});
+const listStyle = Object.freeze({
+    margin: '0',
+});
+const inlineStyle = Object.freeze({
+    display: 'inline-block',
 });
 
-export default function ClaimFacilityIntroStep() {
+function ClaimFacilityIntroStep({ agreement, updateAgreement }) {
+    const [selectedValue, setRadio] = useState(agreement);
+
+    const handleChange = event => {
+        setRadio(event.target.value);
+    };
+
+    useEffect(() => {
+        updateAgreement(selectedValue === 'true');
+    }, [selectedValue]);
```

Comment:
It should be moved to a custom hook. It is best practice to separate presentational and implementation parts.

Reply:
Ok, can be omitted in this case.